modules/standard/Socket.md

```diff
@@ -0,0 +1,210 @@
+  proc nagle(socket:udpSocket, tcpNodeDelay:bool = 1);
+  ```
+- `quickack`
+  This method will set/unset TCP_QUICKACK for provided socket.
+  ```python
+  proc quickack(socket:tcpConn, tcpQuickack:bool = 1)
+  proc qucikack(socket:tcpListener, tcpQuickack:bool = 1);
+  proc quickack(socket:udpSocket, tcpQuickack:bool = 1);
+  ```
+
+## UDP Support
+
+Support for UDP protocol will be through a separate constructor for `udpSocket`. They need not be binded or connected as they are connectionless socket.
+```python
+const socketServ:udpSocket = new udpSocket(family=IPFamily.IPv4)
+```
+
+A generic `bind` function will be used which will take in any of the socket three object and bind it to provided `host` , `port`.
+
+**Method**
+
+```
+proc bind(socketServ, addr:ipAddr, reuseAddr=true)
+```
+
+**Parameters**
+
+```
+- socket : tcpListener|tcpConn|udpSocket - socket object to be bound
+- addr: contains address info about where to bind the socket.
+- reuseaddr: Boolean - optional parameter, default value = true
+```
+
+Bind is responsible for binding any socket to a specified address and port. Here the socket can be anything either a UDP or a TCP Socket. listen under the hood will can in bind to bind `tcpListener` to address and port. With either tcpConn or udpSocket that isn't a requirement as the system assigns any free port and localhost address to them by default.
+
+While `tcpConn` is in a connected state when they are created `udpSocket` are connectionless they are just waiting for the user to use them for sending and receiving purpose only. But for sending and receiving purpose user might require a socket to have a known address that is when bind will be useful for `udpSocket`.
+
+## Data transfer on UDP
+
+UDP Sockets will have method to call `recvFrom` on it this will return the `addr`, `port` of client socket and `data` being written to the socket.
+
+**Method:**
+```python
+proc socketServ.recvFrom() : (addr:ipAddr, data:Generic)
+```
+
+**Return:**
+```
+- addr: ipAddr - contains family info of socket that sent data to our UDP Socket.
+- data: Generic - data to send
+```
+
+Another alternate method will be `recv` which will return just the `data` and no details abouts about client socket.
+
+__Method:__
+```python
+proc socketServ.recv() : (data:Generic)
+```
+
+The `send` method on socket will take in `addr`, `port` and `data` to write to the socket at provided address
+
+__Method:__
+```python
+proc socketServ.send(addr:ipAddr ,data:Generic)
+```
+
+**Parameters:**
+```
+- addr: ipAddr - address info for where to send the data.
+- data: Generic - data to send
+```
+
+## Options on Socket
+
+For providing more granular control over socket options `setSocketOpt` and `getSocketOpt` will be provided which can be used to manipulate socket options. For more information about them [man page](https://man7.org/linux/man-pages/man2/getsockopt.2.html) can be referred as to what options we can set on sockets.
```

Comment:
Reference `https://pubs.opengroup.org/onlinepubs/9699919799/functions/getsockopt.html` instead?

Reply:
I think `man7` is an official page (isn't it ?)

Reply:
The POSIX standard is described in the link Greg provided but the man7 page is the linux manual. We'd like to focus on POSIX rather than linux.

Reply:
ohh right thanks